superset-frontend/src/dashboard/reducers/dashboardState.js

```diff
@@ -132,11 +132,7 @@ export default function dashboardStateReducer(state = {}, action) {
     [TOGGLE_EXPAND_SLICE]() {
       const updatedExpandedSlices = { ...state.expandedSlices };
       const { sliceId } = action;
-      if (updatedExpandedSlices[sliceId]) {
-        delete updatedExpandedSlices[sliceId];
-      } else {
-        updatedExpandedSlices[sliceId] = true;
-      }
+      updatedExpandedSlices[sliceId] = !updatedExpandedSlices[sliceId];
```

Comment:
Without this edit expand_all_slices will trump **Hide chart description** since the expandedSlices key for a chart will be undefined. Not sure if this has any negative impact.

Reply:
If the proposed tests cover this, we'll probably be pretty cozy with it :)